src/vchordrq/algorithm/build.rs

```diff
@@ -203,28 +203,54 @@ impl Structure {
     ) -> Vec<Self> {
         use std::collections::BTreeMap;
         let VchordrqExternalBuildOptions { table } = external_build;
-        let query = format!("SELECT id, parent, vector FROM {table};");
+        let table_name = table.split('.').last().unwrap().to_string();
         let mut parents = BTreeMap::new();
         let mut vectors = BTreeMap::new();
         pgrx::spi::Spi::connect(|client| {
             use crate::datatype::memory_pgvector_vector::PgvectorVectorOutput;
             use base::vector::VectorBorrowed;
             use pgrx::pg_sys::panic::ErrorReportable;
-            let table = client.select(&query, None, None).unwrap_or_report();
-            for row in table {
+            // Get the schema of pgvector
+            let schema_query = format!(
+                "SELECT n.nspname::TEXT 
+                FROM pg_catalog.pg_extension e
+                LEFT JOIN pg_catalog.pg_namespace n ON n.oid = e.extnamespace
+                LEFT JOIN information_schema.columns i ON i.udt_schema = n.nspname
+                WHERE e.extname = 'vector' AND i.table_name = '{table_name}' AND i.column_name = 'vector';");
```

Comment:
Don't reference into system views

No need to reference the table and the column